plugins/panels/model_evaluation.py

```diff
@@ -585,6 +610,89 @@ def load_view(self, ctx):
                     view = eval_view.filter_labels(
                         pred_field, F(eval_key) == field, only_matches=True
                     )
+        elif info.config.type == "segmentation":
+            results = ctx.dataset.load_evaluation_results(eval_key)
```

Comment:
`Dataset` objects do cache evaluation results, so if `load_evaluation_results()` is called multiple times on the same `dataset` singleton instance, the results only have to be loaded from the database once. However, I don't think we can count on multiple API requests re-using the same dataset object...

An alternative would be to store the necessary data as panel state. However that is also not desirable because the necessary data here is potentially huge lists of IDs, and panel state is serialized with every panel callback, not just `load_view()`.

So, the best pattern for this case needs some though.